GOVERNANCE.md

```diff
@@ -0,0 +1,139 @@
+Governance for the R data.table project
+
+# Purpose and scope
+
+## This document
+
+The purpose of this document is to define how people related to the project work together, so that the project can expand to handle a larger and more diverse group of contributors.
+
+## The R package
+
+The purpose of the project is to maintain the R data.table package, which is guided by the following principles:
+
+* No external Imports/LinkingTo/Depends dependencies (external meaning those not maintained by the project)
+* Few (if any) Suggests/Enhances dependencies
+* Time & memory efficiency
+* Concise syntax (minimal redundancy in code)
+* Stable code base (strong preference for user-friendly back-compatibility with data.table itself and with old versions of R)
+* Comprehensive and accessible documentation and run-time signals (errors, warnings)
+
+To prioritize developer time, we define what is in and out of current scope. Feature requests in issues and pull requests that are out of current scope should be closed immediately, because they are not the current priority. If someone wants to contribute code that is currently out of scope, they first have to make a pull request that changes the scope as defined below.
+
+The current scope of package functionality includes:
+* Data manipulation and analysis 
+    * reshaping/pivoting
+    * aggregation/summarizing (via `[,, by=...]` and _grouping sets_)
+    * filtering rows
+    * all sorts of joins
+    * adding/updating/deleting columns
+    * set operations (union/rbind, intersection, difference)
+* high-performance common functions (`frank`, `fcase`, `fifelse`, `transpose`, `chmatch`, `fsort`, `forder`, `uniqueN`, ...)
+* common convenience functions (`%like%`, `%notin%`, `timetaken`, `substitute2`, ...)
+* ordered data functions (`rleid`, `shift`, `fcoalesce`, _locf_/_nocb_ `nafill`, rolling functions)
+* date and time related classes and functions (`IDate`, `ITime`)
+* technical functions (`address`, `tables`, `update_dev_pkg`)
+* Reading/writing of data from/to flat (plain text) files like CSV
+
+Functionality that is out of current scope:
+* Plotting/graphics (like ggplot2)
+* Manipulating out-of-memory data, e.g. data stored on disk or remote SQL DB, (as opposed e.g. to sqldf / dbplyr)
+* Machine learning / modeling (like mlr3)
```

Comment:
```suggestion
* Machine learning (like mlr3)
```
I removed "modeling" because "data modeling" (not statistical modeling) is very much a thing that I was using data.table years before when working on data warehousing projects.